python/core/qgsexpression.sip

```diff
@@ -194,9 +194,9 @@ class QgsExpression
     class Function
     {
       public:
-        Function( QString fnname, int params, QString group, QString helpText = QString(), bool usesGeometry = false, QStringList referencedColumns = QStringList() );
+        Function( const QString& fnname, int params, const QString& group, const QString& helpText = QString(), bool usesGeometry = false, QStringList referencedColumns = QStringList(), bool lazyEval = false );
         /** The name of the function. */
-        QString name();
+        const QString& name();
```

Comment:
AFAIK there's no deep copying. The copy constructor (and assignment operator) of implicitly shared objects always makes a shallow copy. Deep copy is only involved by a call to ```detach()``` which is used by non-const methods. (When using ```QSharedDataPointer```, the non-const ```operator->()``` calls ```detach()``` automatically).

It seems to me that returning ```const QString``` does not have much use:
http://stackoverflow.com/questions/8716330/purpose-of-returning-by-const-value

My preference is ```QString``` for returns, ```const QString&``` for arguments - to be consistent with Qt API